srt/server.go

```diff
@@ -67,31 +66,34 @@ func NewServer(config *Config) Server {
 
 // Listen sets up a SRT socket in listen mode
 func (s *ServerImpl) Listen(ctx context.Context) error {
-	host, portString, err := net.SplitHostPort(s.config.Address)
-	if err != nil {
-		return err
-	}
-
-	if len(host) == 0 {
-		host = "localhost"
-	}
-
-	port, err := strconv.ParseUint(portString, 10, 16)
-	if err != nil {
-		return err
-	}
-
-	addrs, err := net.LookupHost(host)
-	if err != nil {
-		return err
-	}
-
-	for _, address := range addrs {
-		err := s.listenAt(ctx, address, uint16(port))
+	for _, address := range s.config.Addresses {
+		host, portString, err := net.SplitHostPort(address)
 		if err != nil {
 			return err
 		}
-		log.Printf("SRT Listening on %s:%d\n", address, port)
+		if len(host) == 0 {
+		}
+		port, err := strconv.ParseUint(portString, 10, 16)
+		if err != nil {
+			return err
+		}
+		var addrs []string
+		if len(host) != 0 {
+			addrs, err = net.LookupHost(host)
+			if err != nil {
+				return err
+			}
+		} else {
+			addrs = []string{"0.0.0.0", "[::]"}
```

Comment:
> Just for the record: I've done all the requested changes, but when i want to listen on all v6 addresses, i get:
> 
> ```
> srtrelay on  master [!?] via  v1.16.5 
> ❯ ./srtrelay -addresses :1337
> 2021/07/02 15:16:34 Config file not found, using defaults
> 2021/07/02 15:16:34 SRT Listening on 0.0.0.0:1337
> 2021/07/02 15:16:34 Listen failed for [::]:1337 : Error in CreateAddrInet, LookupIP
> ```

```[::]``` is not a valid IP address here as we expect parsed addresses. Also found out that apparently listening on ```::``` is enough to also receive IPv4 connections.